toolkit/tools/internal/network/network.go

```diff
@@ -61,3 +64,23 @@ func DownloadFile(url, dst string, caCerts *x509.CertPool, tlsCerts []tls.Certif
 
 	return
 }
+
+// CheckNetworkAccess checks whether the installer environment has network access
+func CheckNetworkAccess() error {
```

Comment:
Please, let's have a clear separation between function errors and no network access. Otherwise, if this (or any other) function returns `err != nil` you cannot say, if the function failed to do what it was supposed to do, or if the condition we're checking (in this case network access) is not fulfilled. We weren't always consistent with this clear separation but let's try to not make it worse.

How this translates here: the function should return an `error` value and a separate one (I'm guessing a boolean) to say if we've got network connection or not. We should treat `retry.Run()` errors as invalid and use the error from `shell.ExecuteLive` (since we can't really do anything better) to determine if we've got network connection or not.

Reply:
Makes sense to me. Has refactored the implementation